sdks/go/pkg/beam/io/rtrackers/offsetrange/offsetrange_test.go

```diff
@@ -299,3 +301,299 @@ func TestTracker_TrySplit(t *testing.T) {
+	estimator := offsetRangeEndEstimator{EstimateRangeEnd: 0}
+	rest := Restriction{Start: 0, End: math.MaxInt64}
+	tracker, err := NewGrowableTracker(rest, &estimator)
+	if err != nil {
+		t.Fatalf("error creating new GrowableTracker: %v", err)
+	}
+
+	if !tracker.TryClaim(int64(10)) {
+		t.Errorf("tracker.TryClaim(10) = %v, want: %v", false, true)
+	}
+	if !tracker.TryClaim(int64(100)) {
+		t.Errorf("tracker.TryClaim(10) = %v, want: %v", false, true)
+	}
+	if tracker.TryClaim(int64(math.MaxInt64)) {
+		t.Errorf("tracker.TryClaim(math.MaxInt64) = %v, want: %v, %v", true, false, tracker.err)
+	}
+	if tracker.IsDone() {
+		t.Errorf("tracker has done all work, but IsDone() returns false")
+	}
+}
+
+// TestGrowableTracker_SplitBeforeStart tests TrySplit() method for GrowableTracker
+// before claiming anything.
+func TestGrowableTracker_SplitBeforeStart(t *testing.T) {
+	estimator := offsetRangeEndEstimator{EstimateRangeEnd: 0}
+	rest := Restriction{Start: 0, End: math.MaxInt64}
+	tracker, err := NewGrowableTracker(rest, &estimator)
+	if err != nil {
+		t.Fatalf("error creating new GrowableTracker: %v", err)
+	}
+	estimator.SetEstimateRangeEnd(10)
+	p, r, _ := tracker.TrySplit(0)
+
+	expected := Restriction{0, 0}
+	if p.(Restriction) != expected {
+		t.Errorf("wrong primaries after TrySplit(0), got: %v, want: %v", p.(Restriction), expected)
+	}
+	if tracker.GetRestriction().(Restriction) != expected {
+		t.Errorf("wrong restriction tracked by tracker after TrySplit(0), got: %v, want: %v", tracker.GetRestriction().(Restriction), expected)
+	}
+	res := Restriction{0, math.MaxInt64}
+	if res != r.(Restriction) {
+		t.Errorf("wrong residual TrySplit(0), got: %v, want: %v", r.(Restriction), expected)
+	}
+}
+
+// TestGrowableTracker_CheckpointJustStarted tests TryClaim and TrySplit
+// for GrowableTracker.
+func TestGrowableTracker_CheckpointJustStarted(t *testing.T) {
+	estimator := offsetRangeEndEstimator{EstimateRangeEnd: 0}
+	rest := Restriction{Start: 0, End: math.MaxInt64}
+	tracker, err := NewGrowableTracker(rest, &estimator)
+	if err != nil {
+		t.Fatalf("error creating new GrowableTracker: %v", err)
+	}
+	if !tracker.TryClaim(int64(5)) {
+		t.Fatal("tracker.TryClaim(int64(5)) should've claimed.")
+	}
+	estimator.SetEstimateRangeEnd(0)
+	p, r, _ := tracker.TrySplit(0)
+	fmt.Printf("tracker: %#v", tracker)
+	if !tracker.IsDone() {
+		t.Fatal("tracker should have been done, but IsDone() returns false")
+	}
+
+	expPr := Restriction{0, 6}
+	if p.(Restriction) != expPr {
+		t.Errorf("wrong primaries after TrySplit(0), got: %v, want: %v", p.(Restriction), expPr)
+	}
+	if tracker.GetRestriction().(Restriction) != expPr {
+		t.Errorf("wrong restriction tracked by tracker after TrySplit(0), got: %v, want: %v", tracker.GetRestriction().(Restriction), expPr)
+	}
+	expRes := Restriction{6, math.MaxInt64}
+	if r.(Restriction) != expRes {
+		t.Errorf("wrong residual TrySplit(0), got: %v, want: %v", r.(Restriction), expRes)
+	}
```

Comment:
Same thing here, prefer using `got` and `want` to follow the Go idioms.  They can be used as prefixes as well.

It's also possible to inline things in the if to make things cleaer, and easier to use the "correct" variable and avoid copy pasta or repetition errors.

```
if got, want := r.(Restriction), Restriction{5, math.MaxInt64}; got != want {
  t.Errorf( ..., got, want)
}
```

(naturally this isn't possible in all cases, such as the two instances of the primary needing to be checked, so there will be a `want` in the function scope)